components/app-gateway/10-inputs-required.tf

```diff
@@ -84,3 +90,16 @@ variable "app_gateway" {
   })
   description = "Values to use when deploy the app gateway(s)"
 }
+
+variable "certificate" {
+  description = "The certificate to use for the app gateway"
+  type = list(object({
+    gateway_configuration = object({
+      key_vault_name      = string
+      key_vault_secret_id = string
+    })
+    ssl_certificates = list(object({
+      certificate_name = string
+    }))
+  }))
```

Comment:
This should probably just be the type of the `ssl_certificates` property on the `app_gateway` var rather than a new var.